File: datasets/pl-keep-code_davinci_001_temp_0.2/HumanEval_22_filter_integers.pl
# Filter given list of any python values only for integers
# >>> filter_integers(['a', 3.14, 5])
# [5]
# >>> filter_integers([1, 2, 3, 'abc', {}, []])
# [1, 2, 3]
sub filter_integers {
    my($values) = @_;
    my @result;
    for my $value (@$values) {
        push @result, $value if $value =~ /^-?\d+$/;
    }
    return @result;
}

use Test::Deep;


sub testhumaneval {
    my $candidate = \&filter_integers;
        if(eq_deeply($candidate->([]),[])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->([4, {}, [], 23.2, 9, "adasd"]),[4, 9])) {
        print "ok!" }else{
        exit 1;
        }
        if(eq_deeply($candidate->([3, "c", 3, 3, "a", "b"]),[3, 3, 3])) {
        print "ok!" }else{
        exit 1;
        }
}

testhumaneval();

File: datasets/r-keep-code_davinci_001_temp_0.2/HumanEval_22_filter_integers.r
# Filter given list of any python values only for integers
# >>> filter_integers(['a', 3.14, 5])
# [5]
# >>> filter_integers([1, 2, 3, 'abc', {}, []])
# [1, 2, 3]
filter_integers <- function(values) {
    return(list(filter(is.integer, values)))
}
filter_integers(c(1, 2, 3, 4, 5))
filter_integers(c(1, 2, 3, 'abc', {}, []))


test_humaneval <- function() {
candidate <- filter_integers
    if(!identical(candidate(list()), list())){quit('no', 1)}
    if(!identical(candidate(list(4, list(), list(), 23.2, 9, 'adasd')), list(4, 9))){quit('no', 1)}
    if(!identical(candidate(list(3, 'c', 3, 3, 'a', 'b')), list(3, 3, 3))){quit('no', 1)}
}
test_humaneval()